### DISCUSSION TITLE
How do I create a record in node that contains a file?

### DISCUSSION THREAD

**Discussion by kvnlnt** [pocketbase/pocketbase#2343] [Q&A] [answered]
I'm trying to migrate data into pocketbase in node. The docs say that the javascript sdk works in both the [browser and node js.](https://pocketbase.io/docs/client-side-integration/) but when I attempt to create a `FormData` object in node and append a file I get errors. I tried many different iterations but can't seem to figure how to make it work. 

This is my code:
```js
const audioFilePath = `/path/to/${audio.filename}`;
formData.append("audio", fs.createReadStream(audioFilePath), audio.mimetype);
```

This is the error:
```bash
 Failed to execute 'append' on 'FormData': parameter 2 is not of type 'Blob'
 ```
 
If I convert it to a blob with 
```js
 const audioBlob = new Blob([fs.readFileSync(audioFilePath)], {
   type: audio.mimetype ?? "audio/mpeg",
});
```

I get:
```js
response: {
    code: 400,
    message: 'Something went wrong while processing your request.',
    data: { audio: [Object] }
 }
```

What's the correct way to do this? I have 4000 records! 😄

**Comment by ganigeorgiev** [comment, marked as answer]
You can check https://github.com/pocketbase/pocketbase/discussions/1201.

I think the `readFileSync` should work but maybe it is failing a field validation (mimetype or size check). You can console log the `err.response.data.audio` to inspect the field error (or find the failed request in Admin UI > Logs).

**Reply by kvnlnt** [reply]
thanks for the help, took me a few iterations but finally got the validation error as you suggested, I am getting max size warning of. "maximum allowed file size is 5242880 bytes". Sorry to be a pest but I don't see anywhere [in the docs](https://pocketbase.io/docs/files-handling/#uploading-files) to change that limit, can I? I'm uploading an mp3 that's 24mb.

**Reply by kvnlnt** [reply]
nm, found another one of your posts [that answers this](https://github.com/pocketbase/pocketbase/discussions/829). Thanks again, love pocketbase.
